### PR TITLE
Adds checkout screen with validations and creates order

### PR DESCRIPTION
**Pull request by jjszolno**
### Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

---

### Description

Adds checkout form screen with validations and creates order in backend

---

#### Jira board reference

- [Shopping cart: purchase flow](https://www.notion.so/rootstrap/Shopping-cart-purchase-flow-1793f8e26b224f1b86d4d6f6b0b2aaf0?pvs=4)

---

#### Notes:

-

---

#### Tasks:

- [ ] Tested on IOS
- [x] Tested on Android
- [ ] Tested on a small device
- [x] Tested on a real device
- [x] Tested all flows related with this PR changes
- [ ] Tested accessibility
- [ ] Added tests

---

### CODE REVIEW

File: src/network/client/interceptors.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ export default () => {
       return response;
     },
     error => {
+      console.log('error', error.response.data);
```
Comment (guillermomachado) [inline src/network/client/interceptors.ts:45]: 🪵

File: src/features/cart/screen/index.tsx [hunk 2]
```diff
@@ -13,6 +15,7 @@ import CartItem from './cartItem';
 import styles from './styles';
 
 const CartScreen = () => {
+  const { navigate } = useNavigation();
```
Comment (guillermomachado) [inline src/features/cart/screen/index.tsx:18]: Just to remind you the cart screen already receives the `navigate` via props, but tbh I'm 100% hooks

File: src/features/checkout/screen/index.tsx [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+  const [billingAddressLine2, setBillingAddressLine2] = useState('');
+  const [billingAddressLine1, setBillingAddressLine1] = useState('');
+  const [billingPostalCode, setBillingPostalCode] = useState('');
+
+  const months = Array.from({ length: 12 }, (_, i) => ({ label: `${i + 1}`, value: i + 1 }));
+  const currentYear = new Date().getFullYear();
+  const years = Array.from({ length: 20 }, (_, i) => ({
+    label: `${currentYear + i}`,
+    value: currentYear + i,
+  }));
```
Comment (guillermomachado) [inline src/features/checkout/screen/index.tsx:42]: All these + the validation lines in the `handleSubmit` method could be saved by using `react-hook-form` https://react-hook-form.com/

**Review by guillermomachado — APPROVED** [review]

File: src/features/checkout/screen/index.tsx (hunk 3 above)
Comment (jjszolno) [reply src/features/checkout/screen/index.tsx:42]: nice! I'll take a look at it

File: src/features/cart/screen/index.tsx (hunk 2 above)
Comment (jjszolno) [reply src/features/cart/screen/index.tsx:18]: yeah I feel that is much more clean and readable with useNavigation()

**Pull request merged**